### PR TITLE
Finish registration flow

### LINKED ISSUE #6: Complete Registration Flow
Finish the handleSubmit behavior and subsequent pages in the "Register User" flow

### PR DESCRIPTION
**Pull request by alexzhang1618**
Closes #6 
- Implement user registration on clicking 'Sign Up' from /login
  - We don't automatically login in the user, instead navigate back to the login page with a popup
- Note: Passwords need to be > 8 characters, but the login page says at least 8 characters?
  - (this has been addressed)

### CODE REVIEW

File: src/lib/api/AuthAPI.ts [hunk 1]
```diff
@@ -1,6 +1,12 @@
 import { config } from '@/lib';
-import { LoginRequest } from '@/lib/types/apiRequests';
-import { LoginResponse, SendPasswordResetEmailResponse } from '@/lib/types/apiResponses';
+import { LoginRequest, UserRegistration } from '@/lib/types/apiRequests';
+import {
```
Comment (farisashai) [inline src/lib/api/AuthAPI.ts]:
```suggestion
import type {
```

**Comment by alexzhang1618** [comment]
Note: Changed verifyEmail to verify-email, this will need to be adjusted in the backend before v2 is deployed to live

File: src/lib/managers/AuthManager.ts [hunk 2]
```diff
@@ -1,7 +1,11 @@
 import { AuthAPI, UserAPI } from '@/lib/api';
 import { CookieService } from '@/lib/services';
 import { APIHandlerProps } from '@/lib/types';
-import { LoginRequest, SendPasswordResetEmailRequest } from '@/lib/types/apiRequests';
+import {
```
Comment (farisashai) [inline src/lib/managers/AuthManager.ts]:
```suggestion
import type {
```

File: src/lib/managers/AuthManager.ts [hunk 3]
```diff
@@ -25,6 +29,16 @@ export default class AuthManager {
     }
   }
 
+  static async register(data: UserRegistration & APIHandlerProps): Promise<void> {
+    const { onSuccessCallback, onFailCallback, ...userRegistration } = data;
+    try {
+      const user = await AuthAPI.register(userRegistration);
+      onSuccessCallback?.(user);
```
Comment (farisashai) [inline src/lib/managers/AuthManager.ts:36]: Did we end up using this? If not, remove and return `Promise<void>`

**Review by farisashai — CHANGES_REQUESTED** [review]
fix

**Review by farisashai — APPROVED** [review]
LGTM 🚀

**Pull request merged**